app/models/katello/concerns/host_managed_extensions.rb

```diff
@@ -76,6 +79,31 @@ def self.find_with_expiring_pools(_key, _operator, days_from_now)
         end
       end
 
+      def reset_katello_status
+        self.host_statuses.where("#{HostStatus::Status.table_name}.type like :query", query: "Katello%").each do |status|
+          if status.class.const_defined?(:UNKNOWN)
+            status.update!(:status => status.class.const_get(:UNKNOWN))
+          end
+        end
+        self.host_statuses.reload
+        true
+      end
+
+      def reset_subscription_status
+        logger.debug "Scheduling host status cleanup"
+        queue.create(id: "reset_subscription_status_#{id}", name: _("Mark Subscription, and other Status as Unknown for %s") % self,
+          priority: 200, action: [self, :reset_katello_status])
+      end
+
+      def queue_reset_subscription_status
+        should_reset_subscription_status? && reset_subscription_status
```

Comment:
You've been doing too much react development ;)

Reply:
lol